engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/query/SplitFieldsByUnderlyingService.kt

```diff
@@ -0,0 +1,57 @@
+package graphql.nadel.enginekt.transform.query
+
+import graphql.nadel.OperationKind
+import graphql.nadel.Service
+import graphql.nadel.enginekt.schema.FieldCoordinatesToUnderlyingServiceMapping
+import graphql.nadel.enginekt.util.copyWithChildren
+import graphql.nadel.enginekt.util.makeFieldCoordinates
+import graphql.normalized.NormalizedField
+import graphql.normalized.NormalizedQuery
+import graphql.schema.GraphQLSchema
+
+class SplitFieldsByUnderlyingService {
```

Comment:
I think a better name might be `NadelFieldRouter` or something. I know @andimarek hates the term router though, or am I confusing  that with field?

At the very least we aim to prefix all the classes with `Nadel` because it's easier to find. In this case it doesn't matter but I'm sure you've had to sift through 20 `Parser`s when trying to parse a query using GraphQL Java.

<img width="688" alt="image" src="https://user-images.githubusercontent.com/9077461/122858556-908daf80-d36e-11eb-94e8-f97df7de4d5b.png">

Wow maybe 20 wasn't an exaggeration 😂